sprocs/workspace_hosts_drain_extra.sql

```diff
@@ -0,0 +1,23 @@
+CREATE OR REPLACE FUNCTION
+    workspace_hosts_drain_extra(surplus integer) RETURNS void
+AS $$
+BEGIN
+    -- Grab a random assortment of extra hosts
+    CREATE TEMPORARY TABLE extra ON COMMIT DROP AS (
+        SELECT *
+        FROM workspace_hosts AS wh
+        WHERE wh.state = 'ready'
+        ORDER BY random()
```

Comment:
I suggest removing the oldest ones first, so that we tend to cycle in newer hosts. This can provide us with a way to upgrade workspace hosts by launching more with a new AMI version, then scaling down again to kill off the old ones.